chainer/link.py

```diff
@@ -454,6 +457,143 @@ def zerograds(self):
             for param in link._uninitialized_params.values():
                 param._zeroed = True
 
+    def size_num_grads(self):
```

Comment:
I believe that the implementation of `Link` class should be as light as possible because it is one of the core classes in chainer. If the core part contains methods that almost no one will use, they can obscure other useful methods. This method is only for MultiprocessParallelUpdater. We can put this method in the outside of `Link` class, say, in in `multiprocess_parallel_updater.py` as a function. This opinion holds for entire part of this PR in `Link` class. To reflect this opinion will require the modification of many parts, so rebuttal is welcome.